src/services/clients/cxClient.ts

```diff
@@ -156,6 +159,49 @@ export class CxClient {
         return scanResult;
     }
 
+    async updateProjectCustomFields(): Promise<void> {
+        this.log.info("Updating Project Custom Fields.");
+        let projectId = this.projectId;
+        let path = `projects/${projectId}`;
+
+        //reading project custom fields entered by user on UI.
+        const projectCustomFields = this.sastConfig.projectCustomFields.split(",");
+        const projectCustomFieldsIds = new Array<string>(projectCustomFields.length);
+        const projectCustomFieldsKeys = new Array<string>(projectCustomFields.length);
+        const projectCustomFieldsValues = new Array<string>(projectCustomFields.length);
+        for (let i = 0; i < projectCustomFields.length; i++) {
+            projectCustomFieldsKeys[i] = projectCustomFields[i].split(":")[0];
```

Comment:
What if key value delimeter used is not colon character?

Reply:
These are input checks that need to be performed on VSTS code.